### PR TITLE
Fix the parsing error in MSSQL for multiple statements that include `DECLARE` statements

### PR DESCRIPTION
**Pull request by wugeer**
This PR resolves an error in MSSQL that occurs when parsing multiple SQL statements containing `DECLARE` statements.

before
parse sql in mssql
```
declare @a int;
set @a = 2;
select @a * 4;
```
raise error
```
Error during parsing: ParserError("Expected: end of statement, found: set at Line: 2, Column: 1")
```

This resolves issue https://github.com/apache/datafusion-sqlparser-rs/issues/1450

### CODE REVIEW

File: src/parser/mod.rs [hunk 1]
```diff
@@ -5335,7 +5335,7 @@ impl<'a> Parser<'a> {
                 for_query: None,
             });
 
-            if self.next_token() != Token::Comma {
+            if self.peek_token().token == Token::SemiColon || self.next_token() != Token::Comma {
```
Comment (iffyio) [inline src/parser/mod.rs:5338]: Oh could this be simplified if we reused the `self.parse_comma_separated()` variant? I think if thats' possible it would be more conventional to use in the codebase (and would've prevented the edge case)

File: tests/sqlparser_mssql.rs [hunk 2]
```diff
@@ -626,6 +626,72 @@ fn parse_mssql_declare() {
         }],
         ast
     );
+
+    let sql = "DECLARE @bar INT;SET @bar = 2;SELECT @bar * 4";
+    let ast = Parser::parse_sql(&MsSqlDialect {}, sql).unwrap();
```
Comment (iffyio) [inline tests/sqlparser_mssql.rs:631]: Can this use `ms().parse_sql_statements(sql)`?

File: src/parser/mod.rs (hunk 1 above)
Comment (iffyio) [reply src/parser/mod.rs:5338]: @wugeer (just double checking since there was a requested review on this MR), did you get a chance to visit this comment yet?

just in case, to clarify I was thinking as the fix we could just move the body of the loop (minus line) into a method and call that method via parse_comma_separated. for example
```rust
    fn parse_mssql_declare_stmt(&mut self) -> Result<Declare, ParserError> {
        let stmts = self.parse_comma_separated(Parser::parse_mssql_declare_stmt)?;
        Ok(Statement::Declare { stmts })
    }
    fn parse_mssql_declare_stmt(&mut self) -> Result<Declare, ParserError> {
        let name = { ...};
        // ... remaining body verbatim
        let assignment = self.parse_mssql_variable_declaration_expression()?;
        Ok(Declare { names, ... })
    }
```

**Review by iffyio — APPROVED** [review]
LGTM! Thanks @wugeer!
cc @alamb

**Comment by iffyio** [comment]
@wugeer in the meantime when you get the chance could you take a look at the compile issue? i think it likely just needs the latest changes from main incorporated

**Comment by wugeer** [comment]
@iffyio Thank you for your guidance. I have already fixed it. :)

**Comment by alamb** [comment]
Thanks again @wugeer and @iffyio
